src/licensedcode/detection.py

```diff
@@ -263,6 +370,278 @@ def matched_text(
+    2. If this value is a +ve number, i.e. `score`!=`match_coverage`*`rule_Relevance`,
+       then there are extra words in that match.
+
+    :param matched_license: LicenseMatch.
+    """
+    score_coverage_relevance = (
+        license_match.coverage() * license_match.rule.relevance
+    ) / 100
+
+    return score_coverage_relevance - license_match.score()
+
+
+def is_extra_words(license_matches):
+    """
+    Return True if any of the license matches in a file-region has extra words. Having
+    extra words means contains a perfect match with a license/rule, but there are some
+    extra words in addition to the matched text.
+
+    :param license_matches: list
+        List of LicenseMatch.
+    """
+    match_query_coverage_diff_values = (
+        calculate_query_coverage_coefficient(license_match)
+        for license_match in license_matches
+    )
+    return any(
+        match_query_coverage_diff_value > 0
+        for match_query_coverage_diff_value in match_query_coverage_diff_values
+    )
+
+
+def is_false_positive(license_matches):
+    """
+    Return True if all of the license matches in a file-region are false positives.
+    False Positive occurs when other text/code is falsely matched to a license rule,
+    because it matches with a one-word license rule with it's `is_license_tag` value as
+    True. Note: Usually if it's a false positive, there's only one match in that region.
+
+    :param license_matches: list
+        List of LicenseMatch.
+    """
+    start_line_region = min(
+        license_match.start_line for license_match in license_matches
+    )
+    match_rule_length_values = [
+        license_match.rule.length for license_match in license_matches
+    ]
+
+    if start_line_region > FALSE_POSITIVE_START_LINE_THRESHOLD and any(
+        match_rule_length_value <= FALSE_POSITIVE_RULE_LENGTH_THRESHOLD
+        for match_rule_length_value in match_rule_length_values
+    ):
+        return True
+
+    match_is_license_tag_flags = (
+        license_match.rule.is_license_tag for license_match in license_matches
+    )
+    return all(
+        (is_license_tag_flag and match_rule_length == 1)
+        for is_license_tag_flag, match_rule_length in zip(
+            match_is_license_tag_flags, match_rule_length_values
+        )
+    )
+
+
+def has_unknown_matches(license_matches):
+    """
+    Return True if any on the license matches has a license match with an
+    `unknown` rule identifier.
+
+    :param license_matches: list
+        List of LicenseMatch.
+    """
+    return any(match.rule.has_unknown for match in license_matches)
+
+
+def is_unknown_intro(license_match):
+    """
+    Return True if all the license matches is an unknown intro LicenseMatch.
+
+    :param license_matches: list
+        List of LicenseMatch.
+    """
+    return (
+        license_match.rule.has_unknown and
+        license_match.rule.is_license_intro
+    )
+
+
+def is_license_clues(license_matches):
+    """
```

Comment:
Is this instead a method of a LicenseDetection?

Reply:
Same as https://github.com/nexB/scancode-toolkit/pull/2961#discussion_r877480348